### PR TITLE
Added hysteresis to CDL.Integer blocks

### LINKED ISSUE #2978: add hysteresis to blocks in CDL.Integer package
This issue is to add hysteresis to blocks in `CDL.Integer` so the blocks have identical parameters as the ones in `CDL.Continuous` package. Following blocks will be updated:
- [x] `Buildings.Controls.OBC.CDL.Integers.Greater`
- [x] `Buildings.Controls.OBC.CDL.Integers.GreaterThreshold`
- [x] `Buildings.Controls.OBC.CDL.Integers.Less`
- [x] `Buildings.Controls.OBC.CDL.Integers.LessThreshold`

### PR DESCRIPTION
**Pull request by JayHuLBL**
This closes #2978. Added hysteresis to following blocks:
- [x] `Buildings.Controls.OBC.CDL.Integers.Greater`
- [x] `Buildings.Controls.OBC.CDL.Integers.GreaterThreshold`
- [x] `Buildings.Controls.OBC.CDL.Integers.Less`
- [x] `Buildings.Controls.OBC.CDL.Integers.LessThreshold`

### CODE REVIEW

File: Buildings/Controls/OBC/CDL/Integers/Greater.mo [hunk 1]
```diff
@@ -40,31 +147,95 @@ equation
...
+and it switches to <code>false</code> if <i>u<sub>1</sub> &lt; u<sub>2</sub> - h</i>.
+If <i>h = 0</i>, the output is <i>y=u<sub>1</sub> &gt; u<sub>2</sub></i>.
+</p>
+<p>
+Enabling hysteresis can avoid frequent switching.
+Adding hysteresis is recommended in real controllers to guard against sensor noise, and
+in simulation to guard against numerical noise. Numerical noise can be present if
+an input depends on a state variable or a quantity that requires an iterative solution, such as
+a temperature or a mass flow rate of an HVAC system.
+To disable hysteresis, set <code>h=0</code>.
```
Comment (mwetter) [inline Buildings/Controls/OBC/CDL/Integers/Greater.mo:228]: This needs to be deleted as this is not applicable to Integer-valued signals.

File: Buildings/Controls/OBC/CDL/Integers/Greater.mo [hunk 2]
```diff
@@ -11,8 +18,108 @@ block Greater
     "Connector of Boolean output signal"
     annotation (Placement(transformation(extent={{100,-20},{140,20}})));
 
+protected
+  final parameter Boolean have_hysteresis=h >= 1E-10
```
Comment (mwetter) [inline Buildings/Controls/OBC/CDL/Integers/Greater.mo:22]: `h` is an `Integer`, so just check for `h > 0`. (I am surprised that this gives not a warning.) Please also change in the other new blocks.

File: Buildings/Controls/OBC/CDL/Integers/GreaterThreshold.mo [hunk 3]
```diff
@@ -3,15 +3,118 @@ block GreaterThreshold
...
+    annotation (Dialog(tab="Advanced"));
   Interfaces.IntegerInput u
     "Connector of Integer input signal"
     annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
   Interfaces.BooleanOutput y
     "Connector of Boolean output signal"
     annotation (Placement(transformation(extent={{100,-20},{140,20}})));
 
+protected
+  final parameter Boolean have_hysteresis=h >= 1E-10
```
Comment (mwetter) [inline Buildings/Controls/OBC/CDL/Integers/GreaterThreshold.mo:21]: See above, just check for `>0`.

File: Buildings/Controls/OBC/CDL/Integers/GreaterThreshold.mo [hunk 4]
```diff
@@ -39,29 +142,81 @@ equation
...
+and it switches to <code>false</code> if <i>u &lt; t - h</i>.
+If <i>h = 0</i>, the output is <i>y = u &gt; t</i>.
+</p>
+<p>
+Enabling hysteresis can avoid frequent switching.
+Adding hysteresis is recommended in real controllers to guard against sensor noise, and
+in simulation to guard against numerical noise. Numerical noise can be present if
+an input depends on a state variable or a quantity that requires an iterative solution, such as
+a temperature or a mass flow rate of an HVAC system.
+To disable hysteresis, set <code>h=0</code>.
```
Comment (mwetter) [inline Buildings/Controls/OBC/CDL/Integers/GreaterThreshold.mo:209]: Remove, see above

**Comment by JayHuLBL** [comment]
@mwetter The documentation has been updated. Also, the check of `have_hysteresis` have been changed.

**Review by mwetter — CHANGES_REQUESTED** [review]
See inline comments. 
Also, the blocks `Integer.{GreaterEqual,GreaterEqualThreshold}` and `Integer.{LessEqual,LessEqualThreshold}` have no hysteresis. 
I think we should also add it for them so that all inequality comparison, whether strict or weak, have the same parameters, no?

**Comment by mwetter** [comment]
This won't be merged as it add non-needed complexity.

**Pull request closed without merging**